### DISCUSSION TITLE
Working with remote data

### DISCUSSION THREAD

**Discussion by LucaMarconato** [scverse/spatialdata#526] [General]
Working with remote data is currently only partially supported. Here is a collections of related issues and pull requests that constitute a good point to get started from.

Any user request/feedback from users is highly welcomed! 😊 

- https://github.com/scverse/spatialdata/pull/278
- https://github.com/scverse/spatialdata/pull/442
- https://github.com/scverse/spatialdata/issues/441

**Comment by vjcitn** [comment]
What is meant by "consolidated" here in #278 ?

**Comment by LucaMarconato** [comment]
Hi @vjcitn, it refers to putting all the metadata of a Zarr store into a single file at the root level. This is done for two reasons: 1) reduce the number of requests to the remote data; 2) actually be able to discover which data is available in a store for those file systems for which `ls` is not available, such as how it's often the case for S3 object storages.

You can find more details and info in the docs from the [`Zarr-Python` package](https://zarr.readthedocs.io/en/stable/tutorial.html#consolidating-metadata).

**Comment by pakiessling** [comment]
@berombau Hi Benjamin, I am trying our your branch. Really cool work.

```python
import zarr
from spatialdata import SpatialData

s3_address = "https://s3.embl.de/spatialdata/spatialdata-sandbox/merfish.zarr/"
root = zarr.open_consolidated(s3_address, mode="r", metadata_key="zmetadata")
root.info
>... Groups	images, points, shapes, tables ...
sdata = SpatialData.read(root)
sdata
>FileNotFoundError: https://s3.embl.de/spatialdata/spatialdata-sandbox/merfish.zarr/
```
any idea why this is not working?

**Reply by berombau** [reply]
Hi @pakiessling ,

Thanks for trying out the branch. Just to be clear, you tried this with the latest support effort located at the draft PR  https://github.com/scverse/spatialdata/pull/907? This combines a lot of previous PRs that are yet to be merged and on top of those adds remote support for additional elements. 

Can you try just reading from the URL string with SpatialData.read instead of first creating a custom store? It also depends on the storage configuration of the EMBL S3 host and the files accessible there, so maybe also try with a different storage instance of interest to see if that's the cause.

@LucaMarconato

**Reply by pakiessling** [reply]
Ah I was indeed on the wrong branch. Akward..
For `pip install git+https://github.com/berombau/spatialdata@cloud-future` I always get some hatchling non-sense:
` AttributeError: module 'hatchling.build' has no attribute 'prepare_metadata_for_build_editable'`

**Comment by LucaMarconato** [comment]
The most advanced PR for reading data from the cloud is https://github.com/scverse/spatialdata/pull/907, but after adding Zarr v3 support we couldn't merge it right away. This will require more work from our side.
